imsms_analysis/analysis_runner.py

```diff
@@ -73,6 +77,105 @@ def run(self, factory):
+            configs.append(config)
+
+        # Run test
+        configs = [x for x in factory.gen_configurations()]
+
+        self.callbacks.batch_info(configs)
+        for config in configs:
+            try:
+                print(config.analysis_name)
+                
+                # Run this custom preprocessing
+                (
+                final_biom, 
+                target, 
+                _, _
+                ) = run_preprocessing(config, self.callbacks)
+
+                base_dir = "dataset"
+                dataset = "imsms-mlab"
+                preparation = config.analysis_name
+                target_name = "disease_binary"
+                algorithm = config.mlab_algorithm
+                if algorithm is None:
+                    algorithm = "RandomForestClassifier"
+
+                # Create the  expected file structure
+                results_dir = path.join(
+                    base_dir, dataset, preparation, target_name
+                )
+                if not path.exists(results_dir):
+                    makedirs(results_dir)
+                
+                # Save table and metadata
+                table_fp = path.join(
+                    base_dir, dataset, preparation, target_name,
+                    "filtered_table.qza"
+                )
+                table_artifact = Artifact.import_data(
+                    "FeatureTable[Frequency]", 
+                    final_biom
+                )
+                table_artifact.save(table_fp)
+
+                metadata_fp = path.join(
+                    base_dir, dataset, preparation, target_name,
+                    "filtered_metadata.qza"
+                )
+                metadata_artifact = Artifact.import_data(
+                    "SampleData[Target]", target
+                )
+                metadata_artifact.save(metadata_fp)
+                
+
+                # run job indices 1...10 inclusive, letting 5 jobs run at once
+                # each job has 100 parameter sets, for a total of 1000 parameter sets
+                start = 1
+                end = 10
+                n_concurrent_jobs = 5
+                chunk_size = 100
+                (
+                    script_fp,
+                    params_fp,
+                    run_info_fp,
+                ) = orchestrate_hyperparameter_search(
+                    dataset=dataset,
+                    preparation=preparation,
+                    target=target_name,
+                    algorithm=algorithm,
+                    repeats=3,  # num CV repeats
+                    base_dir=base_dir,  # Directory with mlab structure containing datasets
+                    ppn=4,  # processors per node
+                    memory=32,  # memory in GB
+                    wall=50,  # walltime  in hours
+                    chunk_size=chunk_size,  # num parameter sets to run per job
+                    randomize=True,  # randomly shuffle order of parameter set
```

Comment:
Permuting order of features is very similar to changing random seed in RandomForest right?  I suspect that randomization here will change accuracy by around 2-3% in either direction, so if there aren't enough runs to average that away, it may disguise the effects of other factors.  I think I'd prefer everything to be fully deterministic by default at the moment, setting randomize to False, even though I believe this could mask effects of factors that change the feature set.

Reply:
Randomize here means to do a randomized grid search of the hyperparameters - doesn't affect features at all. So if you run the first 50 parameter sets, you are effectively running a randomized grid search of size 50. 
Recommend keeping this on unless you want to run jobs for an algorithm's entire parameter space (RandomForest's entire parameter space is >10,000 parameter sets but other algorithms are much smaller, <1000).

Otherwise, the parameter sets you explore are ordered according to what's returned by SKLearn's ParameterGrid which is basically just itertools.product, and if you run jobs for the first few parameter sets it wouldn't be a random sample of the hyperparameter space.

Reply:
Ahhhhh, understood.  Thought this referred to the columns of the input table.  Should be fine for matching SerialRunner, which only uses one set of parameters then.  Is the randomized grid search seeded in any way?  I would still like to preserve determinism if I run a job with 10000 possible parameter set values but only want to choose 100 of them.  Ideally it would always choose the same 100 if I ran the task three times without specifying a different seed value.

Reply:
Yup, I [seeded it](https://github.com/knightlab-analyses/regression-benchmarking/blob/48cebde806e93be5f5a9ee382e92d14a9b49a6de/q2_mlab/orchestrator.py#L138) for the exact same reason